tofhir-engine/src/main/scala/io/tofhir/engine/data/read/KafkaSourceReader.scala

```diff
@@ -126,19 +126,37 @@ class KafkaSourceReader(spark: SparkSession) extends BaseDataSourceReader[KafkaS
         }).getOrElse(field)
       }).toJson
     })
-
-    spark
-      .readStream
-      .format("kafka")
-      .option("kafka.bootstrap.servers", mappingJobSourceSettings.bootstrapServers)
-      .option("subscribe", mappingSourceBinding.topicName)
-      .option("startingOffsets", mappingSourceBinding.startingOffsets)
-      .option("inferSchema", value = true)
-      .options(mappingSourceBinding.options)
-      .load()
-      .select($"value".cast(StringType)) // change the type of message from binary to string
-      .withColumn("value", processDataUDF(col("value"))) // replace 'value' column with the processed data
-      .select(from_json($"value", schema.get).as("record"))
-      .select("record.*")
+    // Determine whether to use batch or streaming read mode based on the 'batchRead' setting
+    if (mappingJobSourceSettings.batchRead) {
+      spark
+        .read // Use read instead of readStream for a batch read
+        .format("kafka")
+        .option("kafka.bootstrap.servers", mappingJobSourceSettings.bootstrapServers)
+        .option("subscribe", mappingSourceBinding.topicName)
+        .option("startingOffsets", "earliest") // 'earliest' ensures all messages from the topic start point are read;
```

Comment:
I think this is default behavior in spark batch reading. However, we will remove this in another branch so it can stay.

https://spark.apache.org/docs/3.5.1/structured-streaming-kafka-integration.html

Reply:
I've removed it.